stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/typing/DefaultTypingUpdatesBuffer.kt

```diff
@@ -103,10 +103,11 @@ public class DefaultTypingUpdatesBuffer(
      * Should be called before an instance of this class is removed.
      */
     override fun clear() {
+        coroutineScope.cancel()
```

Comment:
The idea here is to kill the scope first so it doesn't trigger a race condition where maybe we post two API calls (automatic & manual one).

Afterwards we flip the boolean which isn't super necessary but is meant to safeguard against clients calling at the wrong time.

Finally we send a manual `onTypingStopped()` call ensuring we send a stop typing event before the screen is exited.